src/app/projects/services/doi.service.ts

```diff
@@ -0,0 +1,138 @@
+import {HttpClient} from '@angular/common/http';
+import {Injectable} from '@angular/core';
+import {forkJoin, Observable, Subject, throwError} from 'rxjs';
+import {map} from 'rxjs/operators';
+import {AutofillProject} from '../models/autofill-project';
+import {EuropePMCHttpSearchResponse, EuropePMCResult, Identifier} from '../models/europe-pmc-search';
+import {Project} from "@shared/models/project";
+import {IngestService} from "@shared/services/ingest.service";
+import {Router} from "@angular/router";
+import {AlertService} from "@shared/services/alert.service";
+
+@Injectable()
+export class DoiService {
+  API_URL = 'https://www.ebi.ac.uk/europepmc/webservices/rest/search';
+  DOI_BASE_URL = 'https://doi.org/';
+
+  loading: Subject<boolean> = new Subject<boolean>();
+
+  constructor(private http: HttpClient,
+              private ingestService: IngestService,
+              private alertService: AlertService,
+              private router: Router) {
+  }
+
+  importProjectUsingDoi(doi: string) {
+    this.loading.next(true);
+    forkJoin({
+      projects: this.getProjectsWithDOI(doi),
+      doiExists: this.doesDoiExist(doi)
+    }).subscribe(({projects, doiExists}) => {
+        this.showExistingProjectsAlert(projects, 'doi');
+
+        if (!doiExists) {
+          this.showDOINotExistsAlert(doi);
+        }
+
+        if (doiExists && projects.length == 0) {
+          this.createProjectWithDoi(doi);
+        }
+
+        this.loading.next(false)
+      },
+      error => {
+        this.alertService.error('An error occurred', error.message);
+        this.loading.next(false);
+      }
+    );
+  }
+
+  getProjectDetails(searchUsing: Identifier, searchString: string): Observable<AutofillProject> {
+    return this.queryEuropePMC(searchUsing, searchString).pipe(
+      map(response => {
+        const result = response.resultList.result[0];
+        if (result) {
+          return this.createAutoFillProject(result);
+        } else {
+          throw throwError(response);
+        }
+      })
+    );
+  }
+  
+  private showExistingProjectsAlert(projects: Project[], publicationIdType: string) {
+    projects.forEach(project => {
+      const projectTitle = project?.content?.['project_core']?.['project_title'];
+      const link = `/projects/detail?uuid=${project?.uuid?.uuid}`;
```

Comment:
using the `?.` syntax means the link may not be correct if the expected object is not present, so the link in the alert won't be correct as well.

Reply:
The link will be broken but that shouldn't happen, all projects should have uuids but it will be displayed successfully, no runtime errors in the UI. I could put a logic to display 'could not generate link, no uuid...'  but i think it's not worth it.